mydemoproject/demoapp1/models/inventory.py

```diff
@@ -6,30 +7,37 @@
 
 class Inventory(BaseModel):
     inventory_id = models.AutoField(primary_key=True)
-    # 0 bcoz we can have 0 books in inventory when all borrowed
-    total_books = models.IntegerField(validators=[MinValueValidator(0)])
+    total_books = models.IntegerField(validators=[MinValueValidator(0)], default=0)
     book = models.OneToOneField(Book, on_delete=models.CASCADE)
 
     def __str__(self):
         return f"{self.book.book_name}: {self.total_books} books available"
 
     @staticmethod
-    def get_total_books(bk):
-        return Inventory.get_all().get(book=bk).total_books
+    def get_total_books(book_id):
+        inventory = Inventory.objects.filter(book_id=book_id).first()
+        if inventory:
+            return True, inventory.total_books
+        else:
+            return False, -1
 
     @classmethod
-    def add_inventory(cls, bk, tbks):
-        invobj= cls(
-            total_books=tbks,
-            book=bk
+    def add_inventory(cls, book, total_books):
+        inventory = cls(
```

Comment:
No check before adding to inventory?

Reply:
We can move the relevant checks here as I have already told you this would cause an exception if someone tries to use it other than your view. Also, there is another check other than total_book < 0 that should be checked here.